pkg/sql/plan2/function/buitins.go

```diff
@@ -971,6 +971,13 @@ var builtins = map[int]Functions{
 				Args:      []types.T{types.T_varchar, types.T_int64, types.T_varchar},
 				ReturnTyp: types.T_varchar, Fn: multi.Lpad,
 			},
+			{
```

Comment:
there is no need to register an overload
{varchar, varchar, varchar}

in fact, the implicit type convert logic for function's argument will do the work for you.
I will try to convert char or varchar to int64.

and if you don't want to use the logic, you can write a special function for the LPAD function.
just put it at Functions.TypeCheckFn